sympy/solvers/solveset.py

```diff
@@ -416,7 +419,12 @@ def _solve_trig(f, symbol, domain):
     elif solns is S.EmptySet:
         return S.EmptySet
     else:
-        return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
+        #try solving through solve_decomposition
+        try:
+            result = solve_decomposition(f_original, symbol, domain)
```

Comment:
I was wondering if we use `solve_decomposition` while implementing `transolve`. Probably we can use it when `transolve` is not able to solve equations.
I am not sure but would that be a better idea?